modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -178,9 +184,55 @@ class ONNXImporter
     const std::string str_domain_ai_onnx = "ai.onnx";
 };
 
+struct LayerPayload
+{
+    std::string layer_type_domain;
+
+    bool operator<(const LayerPayload& rhs) const
+    {
+        return layer_type_domain < rhs.layer_type_domain;
+    }
+};
+
+std::ostream& operator<<(std::ostream& os, const LayerPayload& payload)
+{
+    os << "domain: " << payload.layer_type_domain;
+    return os;
+}
+
+class ONNXLayerHandler : public detail::PayloadedLayerHandler<LayerPayload>
+{
+public:
+    explicit ONNXLayerHandler(ONNXImporter* importer_);
+
+    void fillRegistry(const opencv_onnx::GraphProto& net);
+
+protected:
+    ONNXImporter* importer;
+    const std::string str_domain_ai_onnx = "ai.onnx";
+};
+
+ONNXLayerHandler::ONNXLayerHandler(ONNXImporter* importer_) : importer(importer_){}
+
+void ONNXLayerHandler::fillRegistry(const opencv_onnx::GraphProto &net)
+{
+    int layersSize = net.node_size();
+    for (int li = 0; li < layersSize; li++) {
+        const opencv_onnx::NodeProto &node_proto = net.node(li);
+        const std::string& name = node_proto.output(0);
+        const std::string& type = node_proto.op_type();
+        const std::string& layer_type_domain = importer->getLayerTypeDomain(node_proto);
+        const auto& dispatch = importer->getDispatchMap(node_proto);
+        if (dispatch.find(type) == dispatch.end())
+        {
+            addMissing(type, name, LayerPayload{layer_type_domain});
```

Comment:
`domain` is some kind of `type` part (like namespace).
No sense to use/handle it separately as semantic and versioning of the same operation from different domain may be different.